File: extensions/dotNetClientAPI/eccCommonDataModel-Monitor/EMPostReportSummary.cs
/////////////////////////////////////////////////////////////////////////
//
// © University of Southampton IT Innovation Centre, 2012
//
// Copyright in this software belongs to University of Southampton
// IT Innovation Centre of Gamma House, Enterprise Road, 
// Chilworth Science Park, Southampton, SO16 7NS, UK.
//
// This software may not be used, sold, licensed, transferred, copied
// or reproduced in whole or in part in any manner or form or in or
// on any media by any person other than in accordance with the terms
// of the Licence Agreement supplied with the software, or otherwise
// without the prior written consent of the copyright owners.
//
// This software is distributed WITHOUT ANY WARRANTY, without even the
// implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
// PURPOSE, except where stated in the Licence Agreement supplied with
// the software.
//
//      Created By :            Simon Crowle
//      Created Date :          09-Apr-2013
//      Created for Project :   EXPERIMEDIA
//
/////////////////////////////////////////////////////////////////////////

using uk.ac.soton.itinnovation.experimedia.arch.ecc.common.dataModel.metrics;

using System;
using System.Collections.Generic;




namespace uk.ac.soton.itinnovation.experimedia.arch.ecc.common.dataModel.monitor
{

    /**
     * EMPostReportSummary encapsulates Reports of MeasurementSets describing the
     * range of the data generated by the user client during the course of the experiment.
     * 
     * @author sgc
     */
    public class EMPostReportSummary
    {
        private Dictionary<Guid, Report> reportsByMeasurementSetID;

        public EMPostReportSummary()
        {
            reportsByMeasurementSetID = new Dictionary<Guid, Report>();
        }

        /**
         * Returns a set of all the MeasurementSet IDs referred to by this summary report.
         * 
         * @return - Set of IDs for the MeasurementSets in this summary
         */
        public ICollection<Guid> getReportedMeasurementSetIDs()
        { return reportsByMeasurementSetID.Keys; }

        /**
         * Adds a report to this summary. The report instance should contain correct
         * references to the MeasurementSet IDs.
         * 
         * @param report - Report instance (fields must not be null)
         */
        public void addReport(Report report)
        {
            if (report != null)
            {
                MeasurementSet ms = report.measurementSet;

                if (ms != null)
                    reportsByMeasurementSetID.Add(ms.msetID, report);
            }
        }

        /**
         * Removes the Report associated with the MeasurementSet ID
         * 
         * @param measurementSetID - ID of the MeasurementSet
         */
        public void removeReport(Guid measurementSetID)
        {
            if (measurementSetID != null)
                reportsByMeasurementSetID.Remove(measurementSetID);
        }

        /**
         * Gets the report associated with the MeasurementSet ID
         * 
         * @param measurementID - MeasurementSet ID
         * @return              - Report instance detailing the metric data for the MeasurementSet ID
         */
        public Report getReport(Guid measurementID)
        {
            Report report = null;

            if (measurementID != null)
                report = reportsByMeasurementSetID[measurementID];

            return report;
        }
    }

} // namespace

File: extensions/cppNetClientAPI/eccCommonDataModel/Monitor/EMPostReportSummary.cpp
/////////////////////////////////////////////////////////////////////////
//
// © University of Southampton IT Innovation Centre, 2012
//
// Copyright in this software belongs to University of Southampton
// IT Innovation Centre of Gamma House, Enterprise Road, 
// Chilworth Science Park, Southampton, SO16 7NS, UK.
//
// This software may not be used, sold, licensed, transferred, copied
// or reproduced in whole or in part in any manner or form or in or
// on any media by any person other than in accordance with the terms
// of the Licence Agreement supplied with the software, or otherwise
// without the prior written consent of the copyright owners.
//
// This software is distributed WITHOUT ANY WARRANTY, without even the
// implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
// PURPOSE, except where stated in the Licence Agreement supplied with
// the software.
//
//      Created By :            Simon Crowle
//      Created Date :          21-June-2013
//      Created for Project :   EXPERIMEDIA
//
/////////////////////////////////////////////////////////////////////////

#include "stdafx.h"
#include "EMPostReportSummary.h"


using namespace boost::uuids;
using namespace std;


namespace ecc_commonDataModel
{

EMPostReportSummary::EMPostReportSummary()
  : ModelBase()
{
}

EMPostReportSummary::~EMPostReportSummary()
{
}

hash_set<uuid> EMPostReportSummary::getReportedMeasurementSetIDs()
{
  hash_set<uuid> reportIDs;

  // Copy UUIDs into a collection
  ReportMap::iterator repIt = reportsByMeasurementSetID.begin();
  while ( repIt != reportsByMeasurementSetID.end() )
  {
    reportIDs.insert( repIt->first );
    ++repIt;
  }

  return reportIDs;
}

void EMPostReportSummary::addReport( Report::ptr_t report )
    
{
  if ( report )
  {
    MeasurementSet::ptr_t ms = report->getMeasurementSet();

    if ( ms )
      reportsByMeasurementSetID.insert( make_pair<uuid,Report::ptr_t>( ms->getMSetID(), report ) );
  }
}

void EMPostReportSummary::removeReport( const uuid& measurementSetID )
{
  reportsByMeasurementSetID.erase( measurementSetID );
}

Report::ptr_t EMPostReportSummary::getReport( const uuid& measurementID )
{
  Report::ptr_t report;

  ReportMap::iterator target = reportsByMeasurementSetID.find( measurementID );

  if ( target != reportsByMeasurementSetID.end() )
    report = target->second;

  return report;
}

// ModelBase -----------------------------------------------------------------
void EMPostReportSummary::toJSON( wstring& jsonStrOUT )
{
}

void EMPostReportSummary::fromJSON( const wstring& jsonStr )
{
}

wstring EMPostReportSummary::toString()
{
  wstring ts;

  return ts;
}

} // namespace
